test/tools/test_tools_memories.py

```diff
@@ -58,3 +62,412 @@ def test_sopel_identifier_memory_channel_str():
     assert memory['#adminchannel'] == test_value
     assert memory['#AdminChannel'] == test_value
+
+
+def test_sopel_identifier_memory_setdefault():
+    memory = memories.SopelIdentifierMemory()
+    assert 'foo' not in memory
+
+    memory.setdefault('Foo')
+    assert len(memory) == 1
+    assert 'foo' in memory
+    assert 'fOO' in memory
+    assert memory['fOo'] is None
+
+    memory.setdefault('bAr', 'DEFAULT')
+    assert len(memory) == 2
+    assert 'FoO' in memory
+    assert 'BaR' in memory
+    assert memory['Bar'] == 'DEFAULT'
+
+    assert memory.setdefault('baR', 'different') == 'DEFAULT'
+    assert len(memory) == 2  # still, that is
+
+
+def test_sopel_identifier_memory_copy():
+    memory = memories.SopelIdentifierMemory()
+    memory['SomeCamelCase'] = True
+    memory['loweronly'] = False
+    assert len(memory) == 2
+    assert type(memory).__name__ == 'SopelIdentifierMemory'
+
+    copied = memory.copy()
+    assert len(copied) == 2
+    assert 'SomeCamelCase' in copied
+    assert 'loweronly' in copied
+    assert type(copied).__name__ == 'SopelIdentifierMemory'
+
+
+def test_sopel_identifier_memory_get():
+    memory = memories.SopelIdentifierMemory()
+    memory['SomeCamelCase'] = True
+    memory['loweronly'] = False
+    assert len(memory) == 2
+
+    # verify key-exists behavior w/implicit default
+    got = memory.get('somecamelcase')
+    assert got is True
+    del got
+    got = memory.get('LowerOnly')
+    assert got is False
+    del got
+
+    # verify key-exists behavior w/explicit default of None
+    got = memory.get('somecamelcase', None)
+    assert got is True
+    del got
+    got = memory.get('LowerOnly', None)
+    assert got is False
+    del got
+
+    # verify key-exists behavior w/explicit "real" default value
+    got = memory.get('somecamelcase', 'DEFAULT')
+    assert got is True
+    del got
+    got = memory.get('LowerOnly', 'DEFAULT')
+    assert got is False
+    del got
+
+    # verify key-missing behavior w/implicit default
+    assert 'missing_key' not in memory
+    got = memory.get('missing_key')
+    assert got is None
+    del got
+
+    # verify key-missing behavior w/explicit default of None
+    assert 'missing_key' not in memory
+    got = memory.get('missing_key', None)
+    assert got is None
+    del got
+
+    # verify key-missing behavior w/explicit "real" default value
+    assert 'missing_key' not in memory
+    got = memory.get('missing_key', 'DEFAULT')
+    assert got == 'DEFAULT'
+
+
+def test_sopel_identifier_memory_pop():
+    memory = memories.SopelIdentifierMemory()
+    memory['SomeCamelCase'] = True
+    assert len(memory) == 1
+
+    # verify SopelIdentifierMemory.pop('key') behavior
+    pop_me = memory.copy()
+    assert len(pop_me) == 1
+
+    popped = pop_me.pop('someCAMELcase')
+    assert len(memory) == 1  # sanity check
+    assert len(pop_me) == 0
+    assert popped is True
+
+    # reset
+    del pop_me
+    del popped
```

Comment:
I similarly hate the `del` here but hopefully the rewrite for the `get()` test gets that across well enough that you can see what I would have written here instead.